src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -214,6 +223,12 @@ void ErrorHandler(object sender, Newtonsoft.Json.Serialization.ErrorEventArgs ev
                         }
                     }
 
+                    if (!(exception is JsonException || exception is OverflowException))
```

Comment:
Why is `OverflowException` special? Could you write a comment about it?

Reply:
This exception happens in certain cases like here (where the value of '300' is invalid value for a byte...notice the array here), but the if the same thing were inside a nested object then we get a JsonReaderException, so not sure what comment i have to write here. https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs#L341

Reply:
I guess you can say that we'll auto handle all value conversion errors? Seems like that was the intent of https://github.com/aspnet/Mvc/commit/40b7636b720298a06b9a42017a8bcace34132fad